cpp/src/arrow/compute/exec/asof_join_benchmark.cc

```diff
@@ -0,0 +1,171 @@
+    AsyncGenerator<util::optional<ExecBatch>> sink_gen;
+    MakeExecNode("sink", plan.get(), {join_node}, SinkNodeOptions{&sink_gen});
+    state.ResumeTiming();
+    ASSERT_FINISHES_OK(StartAndCollect(plan.get(), sink_gen));
+  }
+
+  state.counters["total_rows_per_second"] = benchmark::Counter(
+      static_cast<double>(state.iterations() *
+                          (left_table_stats.total_rows + right_hand_rows)),
+      benchmark::Counter::kIsRate);
+
+  state.counters["total_bytes_per_second"] = benchmark::Counter(
+      static_cast<double>(state.iterations() *
+                          (left_table_stats.total_bytes + right_hand_bytes)),
+      benchmark::Counter::kIsRate);
+
+  state.counters["maximum_peak_memory"] =
+      benchmark::Counter(static_cast<double>(ctx.memory_pool()->max_memory()));
+}
+
+static void AsOfJoinOverhead(benchmark::State& state) {
+  int64_t tolerance = 0;
+  AsofJoinNodeOptions options = AsofJoinNodeOptions(kTimeCol, kKeyCol, tolerance);
+  TableJoinOverhead(
+      state,
+      TableGenerationProperties{int(state.range(0)), int(state.range(1)),
+                                int(state.range(2)), "", kDefaultMinColumnVal,
+                                kDefaultMaxColumnVal, 0, kDefaultStart, kDefaultEnd},
+      int(state.range(3)),
+      TableGenerationProperties{int(state.range(5)), int(state.range(6)),
+                                int(state.range(7)), "", kDefaultMinColumnVal,
+                                kDefaultMaxColumnVal, 0, kDefaultStart, kDefaultEnd},
+      int(state.range(8)), int(state.range(4)), "asofjoin", options);
+}
+
+// this generates the set of right hand tables to test on.
+void SetArgs(benchmark::internal::Benchmark* bench) {
+  bench
+      ->ArgNames({"left_freq", "left_cols", "left_ids", "left_batch_size",
+                  "num_right_tables", "right_freq", "right_cols", "right_ids",
+                  "right_batch_size"})
+      ->UseRealTime();
+
+  int default_freq = 5;
+  int default_cols = 20;
+  int default_ids = 500;
+  int default_num_tables = 1;
+  int default_batch_size = 100;
+
+  for (int freq : {1, 5, 10}) {
+    bench->Args({freq, default_cols, default_ids, default_batch_size, default_num_tables,
+                 freq, default_cols, default_ids, default_batch_size});
+  }
+  for (int cols : {10, 20, 100}) {
+    bench->Args({default_freq, cols, default_ids, default_batch_size, default_num_tables,
+                 default_freq, cols, default_ids, default_batch_size});
+  }
+  for (int ids : {100, 500, 1000}) {
+    bench->Args({default_freq, default_cols, ids, default_batch_size, default_num_tables,
+                 default_freq, default_cols, ids, default_batch_size});
+  }
+  for (int num_tables : {1, 10, 50}) {
+    bench->Args({default_freq, default_cols, default_ids, default_batch_size, num_tables,
+                 default_freq, default_cols, default_ids, default_batch_size});
+  }
+  for (int batch_size : {1, 500, 1000}) {
```

Comment:
These are extremely small values of `batch_size`.  Currently, when scanning from files, we use `32Ki`.  `1000` is probably a lower bound.  Can you use `1Ki`, `4Ki`, and `32Ki` instead?

Reply:
Increased batch size, also increased the data generation time range so this batch size makes (more) sense.